cachi2/core/package_managers/npm.py

```diff
@@ -476,6 +502,104 @@ def _get_npm_dependencies(
     return download_paths
 
 
+def get_type_of_dependency(dependency_version: str) -> Literal["registry", "git", "file", "https"]:
+    """Return type of dependency.
+
+    :param dependency_version: Dependency version loaded from package(-lock).json dependencies
+    """
+    url = urlparse(dependency_version)  # TODO: Do we need to support also http: ? I believe not
+    if url.scheme == "file":
+        return "file"
+    elif (
+        url.scheme == "git"
+        or url.scheme.startswith("git+")
+        or (url.scheme == "https" and url.path.endswith(".git"))
+    ):
+        return "git"
+    elif url.scheme == "https" and url.path.endswith(".tar.gz") or url.path.endswith(".tgz"):
+        return "https"
+    elif url.path.count("/") == 1:
+        # GitHub URLs are also supported in form of namespace/project e.g. "expressjs/express"
+        return "git"
+    else:
+        # Anything else should be considered as registry version or range of registry versions or latest
+        return "registry"
+
+
+def update_package_lock_with_local_paths(
+    download_paths: Dict[NormalizedUrl, RootedPath],
+    package_lock_file: ProjectFile,
+    output_dir: RootedPath,
+) -> ProjectFile:
+    """Replace packages resolved URLs with local paths.
+
+    Update package-lock.json file in a way it can be used in isolated environment (container)
+    without internet connection. All package/dependency URLs will be replaced for
+    local paths to downloaded dependencies.
+
+    :param download_paths:
+    :param package_lock_file: File path to package-lock.json
+    :param output_dir: Output directory with fetched dependencies
+    """
+    package_lock = PackageLock.from_file(RootedPath(package_lock_file.abspath))
```

Comment:
Do we need to re-parse the package-lock.json? Can we update the PackageLock instance that we already have?

I think the idea was that the `info["package_lock_file"]` returned from resolve_npm would already be the updated file

Reply:
You're right. We don't. Main logic was moved to `_resolve_npm`